### DISCUSSION TITLE
ollamaApiClient.StreamCompletion no longer there?

### DISCUSSION THREAD

**Discussion by chrisnieboer** [awaescher/OllamaSharp#151] [General]
I have seen quite a few examples where the OllamaApiClient has a method all StreamCompletion. Was this deprecated in a later release? if so what is the proper way to send a request and have it return a ConversationContext?

**Comment by awaescher** [comment]
Yes, with [version 3, I streamlined the method names to match the Ollama API](https://github.com/awaescher/OllamaSharp/releases/tag/3.0.0).

Using the `context` [has been deprecated by the Ollama team](https://github.com/ollama/ollama/pull/7878) in favor for the `/chat` endpoint for which I would recommend using the `Chat` class which is automatically tracking a message history for you.

``` csharp
// set up the client
var uri = new Uri("http://localhost:11434");
var ollama = new OllamaApiClient(uri);

// select a model which should be used for further operations
ollama.SelectedModel = "llama3.1:8b";

var chat = new Chat(ollama);
while (true)
{
    var message = Console.ReadLine();
    await foreach (var answerToken in chat.SendAsync(message))
        Console.Write(answerToken);
}
// messages including their roles and tool calls will automatically be tracked within the chat object
// and are accessible via the Messages property
```
